### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by joeyGold1**

### CODE REVIEW

File: app/gilded-rose.ts [hunk 1]
```diff
@@ -10,60 +10,133 @@ export class Item {
...
+    return quality>0 ? quality - 1 : quality ;
+  }
+
+  const updatedSulfurasQuality = (quality) => {
+    return quality!==80?quality=80:quality;    
+  }
+
+  const updateSellIn = (sellIn)=>{
+    return sellIn-1;
+  }
```
Comment (joeyGold1) [inline app/gilded-rose.ts]: Generally when writing javascript it's better practice not to write function definitions inside another function definition as you've done with these. It's not a big issue and it will work fine for this but could cause some performance issues

File: app/gilded-rose.ts [hunk 2]
```diff
@@ -10,60 +10,133 @@ export class Item {
...
+         
+        for (const item of this.items) {
```
Comment (joeyGold1) [inline app/gilded-rose.ts:63]: Careful with indentation levels here - this has jumped 4 levels when the indentation didn't need to change at all

File: app/gilded-rose.ts [hunk 3]
```diff
@@ -10,60 +10,133 @@ export class Item {
...
+            case 'Sulfuras, Hand of Ragnaros': 
+              item.quality = updatedSulfurasQuality(item.quality);  
+              //return item
+              break;
+
+            default:
+              item.quality = updatedItemQuality(item.quality, item.sellIn);
+              item.sellIn = updateSellIn(item.sellIn);
+             //return item
+              break;
```
Comment (joeyGold1) [inline app/gilded-rose.ts:86]: Good use of a switch/case here!

File: app/gilded-rose.ts [hunk 4]
```diff
@@ -10,60 +10,133 @@ export class Item {
...
+
+            case 'Aged Brie':
+              item.quality = updateAgedBrieQuality(item.quality);
+              item.sellIn = updateSellIn(item.sellIn); 
+              break;     
+
+            case 'Backstage passes to a TAFKAL80ETC concert':
+              item.quality = updateBackstagePassesQuality(item.quality, item.sellIn);
+              item.sellIn = updateSellIn(item.sellIn);
+              //return item
```
Comment (joeyGold1) [inline app/gilded-rose.ts:74]: Try not to leave commented out code in when you commit

File: app/gilded-rose.ts [hunk 5]
```diff
@@ -10,60 +10,133 @@ export class Item {
...
+    }
+  }
+}
+
+return this.items;
+
+
+*/
 }
+}
```
Comment (joeyGold1) [inline app/gilded-rose.ts:148]: Same here - there's no need to leave old code around. It's stored in git if we ever need it again

File: test/jest/gilded-rose.spec.ts [hunk 6]
```diff
@@ -1,9 +1,103 @@
...
+    expect(items[0].sellIn).toBe(9);
+    expect(items[0].quality).toBe(50);
+  });
+
+  it('`Quality` of "Backstage passes" is never more than 50', () => {
+    const gildedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 10, 50)]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].sellIn).toBe(9);
+    expect(items[0].quality).toBe(50);
+  });
```
Comment (joeyGold1) [inline test/jest/gilded-rose.spec.ts:55]: Really good tests - sticking to the given, when, then process really well!

**Review by joeyGold1 — COMMENTED** [review]
Nice code and tests!

Only thing I'd like to see is more tests around edge cases - what happens if quality is 49 and we try and add 2 for example?
